### PR TITLE
Fix for google one tap failure

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Proposed changes in this pull request

$subject

### CODE REVIEW

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java [hunk 1]
```diff
@@ -534,7 +534,7 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
             if (nonce == null) {
                 LOG.debug("OIDC provider does not support nonce claim in id_token.");
             }
-            if (nonce != null && !nonce.equals(context.getProperty(NONCE))) {
+            if (nonce != null && context.getProperty(NONCE) != null && !nonce.equals(context.getProperty(NONCE))) {
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:537]: Lets add the null check before the `if (nonce == null) {` line. So the complete nonce validation portion will only engage if the nonce is available in the context

Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java]: Fixed by - https://github.com/wso2-extensions/identity-outbound-auth-oidc/pull/142/commits/014c26a150f1306923f029c49350e98e7c039f18

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java [hunk 2]
```diff
@@ -530,13 +530,15 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
...
             }
 
-            String nonce = (String) jwtAttributeMap.get(NONCE);
-            if (nonce == null) {
-                LOG.debug("OIDC provider does not support nonce claim in id_token.");
-            }
-            if (nonce != null && !nonce.equals(context.getProperty(NONCE))) {
-                throw new AuthenticationFailedException(ErrorMessages.NONCE_MISMATCH.getCode(),
-                        ErrorMessages.NONCE_MISMATCH.getMessage());
+            if (context.getProperty(NONCE) != null) {
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:533]: Saw that we are setting property to the context from the constant org.wso2.carbon.identity.application.authenticator.oidc.OIDCAuthenticatorConstants.Claim.NONCE which is simply "nonce". 

Can we introduce a new constant for this and make it specific for the authenticator. something like "OIDC_FEDERATION_NONCE" (you can come up with a better name). The reason is the authentication context is used in various places and we are using a general property map to store this, so if any other component decided that its also going to store a nonce then we will start seeing erroneous results here. So its good to have a unique property name that is unlikely to be used by some other component.

Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:533]: +1. Thanks for the suggestion and will send an improvement

Comment (janakamarasena) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java:533]:
```suggestion
            if (StringUtils.isNotBlank(NONCE)) {
```

Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java]: Fixed by - https://github.com/wso2-extensions/identity-outbound-auth-oidc/pull/142/commits/1e00ef0a95ea6894616780d0712fff8d89475f7a

**Pull request merged**